src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/RemappedLevelLogger.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    internal class RemappedLevelLogger : ILogger
```

Comment:
I think if you implement https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/Logging/LoggerBase.cs you may just need to implement 2 methods, but i might be missing something.